crates/menmos-std/src/sync/sharded_mutex.rs

```diff
@@ -0,0 +1,70 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+use tokio::sync::{RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+// See https://preshing.com/20110504/hash-collision-probabilities/
+fn optimal_bucket_count(concurrent_calls: usize, mut collision_probability: f64) -> usize {
+    if collision_probability == 0.0 {
+        collision_probability = f64::EPSILON;
```

Comment:
Why epsilon? Can you add a comment.

Reply:
epsilon is the smallest representable floating point number, to use instead of zero. I'll add a small comment